Framework/Kernel/src/DateAndTimeHelpers.cpp

```diff
@@ -93,14 +93,46 @@ averageSorted(const std::vector<Types::Core::DateAndTime> &times) {
   // to avoid overflow subtract the first time off from everything
   // and find the average in between
   const int64_t first = times.begin()->totalNanoseconds();
-  int64_t total = 0;
-  for (const auto &it : times)
-    total += (it.totalNanoseconds() - first);
+  int64_t total =
+      std::accumulate(times.begin(), times.end(), int64_t{0},
+                      [first](int64_t a, const Types::Core::DateAndTime time) {
+                        return std::move(a) + (time.totalNanoseconds() - first);
```

Comment:
Why the `std::move`?

Reply:
Because it was in the example for [`std::accumulate`](https://en.cppreference.com/w/cpp/algorithm/accumulate)